technic/radiation.lua

```diff
@@ -544,8 +561,17 @@ if griefing then
 				vector.new(0,0,-1),
 				vector.new(0,-1,0),
 			}) do
-				if math.random(8) == 1 then
-					minetest.dig_node(vector.add(pos, offset))
+				local vpos = vector.add(pos, offset)
+				local vnode = minetest.get_node(vpos)
+				local can_dig = true
+				if vnode and vnode.name and minetest.registered_nodes[vnode.name] then
+					local hardness = minetest.registered_nodes[vnode.name]._mcl_hardness
+					if hardness and type(hardness) == "number" then
+						can_dig = hardness > 0
+					end
+				end
+				if math.random(8) == 1 and can_dig then
+					minetest.dig_node(vpos)
```

Comment:
```suggestion
				if math.random(8) == 1 then
					local vpos = vector.add(pos, offset)
					local node = minetest.get_node(vpos)
					if minetest.get_item_group(node.name, "_mcl_hardness") > 0 then
						minetest.dig_node(vpos)
					end
```

Reply:
_mcl_hardness is not a group, is a node def param, also this mode skip the sec check, if node is unknow node, this crashs

Reply:
> _mcl_hardness is not a group, is a node def param

Right, I forgot about that... 🤦 It really should be a group though.

> also this mode skip the sec check, if node is unknow node, this crashes

Not exactly, `minetest.get_node` always returns a value, so there's no need to check that, but you do have to check if the definition exists.

Revised suggestion:
```suggestion
				if math.random(8) == 1 then
					local vpos = vector.add(pos, offset)
					local def = minetest.registered_nodes[minetest.get_node(vpos).name]
					if def and (not def._mcl_hardness or def._mcl_hardness > 0) then
						minetest.dig_node(vpos)
					end
```